libraries/AP_Motors/AP_MotorsHeli_RSC.cpp

```diff
@@ -210,3 +212,53 @@ float AP_MotorsHeli_RSC::calculate_desired_throttle(float collective_in)
 
 }
 
+void AP_MotorsHeli_RSC::estimate_rpm()
+{
+    const uint8_t imu_instance = 0;
+    const AP_InertialSensor &ins = AP::ins();
+    const Vector3f &accel = ins.get_accel(imu_instance);
+//    uint64_t now = AP_HAL::micros64();
+
+    if (sample_cnt < 2*fftLen) {
+//        p[sample_cnt] = accel.y + 0.2 * sinf(6.28e-6f*40.0f*now);
+        p[sample_cnt] = accel.y;
+        estimated_rpm = cfft[sample_cnt/2];
+        sample_cnt++;
+        p[sample_cnt] = 0.0f;
+        sample_cnt++;
+    } else {
+        sample_cnt = 0;
+        //run fft
+        if (arm_cfft_radix4_init_f32( fft, fftLen, 0, 1) == ARM_MATH_SUCCESS) {
```

Comment:
I'm guessing you're satisfied we're not going to stop flying the aircraft for too long while doing all this :-)

Any timings-on-hardware you could put in as a comment, by any chance?

Reply:
Not sure what you mean about comments on hardware.  Not sure I know enough to make a judgement like that.  As long as we can get accelerometer data at 400 hz, this should work.  I'm thinking of making this 1000hz in order to get the rotor speed resolution to 1 hz.  Also thinking of  doing this twice a second which would require staggering fft's.

Reply:
Also this was just for testing the code.  I plan on using the RPM library for the FFT estimate of RPM.   So this commit will also be deleted before this gets merged.

Reply:
Comment would looks something like https://github.com/ardupilot/ardupilot/blob/master/libraries/GCS_MAVLink/GCS_Common.cpp#L1014

You can, perhaps, see how I gather that information in the GCS library; disable interrupts, take a start time, run fft, take stop time, re-enable interrupts.  Put comment in code.  Reconsider code if it comes in at 2000 microseconds :-)